src/main/kotlin/com/jetbrains/snakecharm/framework/SmkDisabledBanner.kt

```diff
@@ -0,0 +1,70 @@
+
+import com.intellij.icons.AllIcons
+import com.intellij.ide.util.PropertiesComponent
+import com.intellij.openapi.fileEditor.FileEditor
+import com.intellij.openapi.fileEditor.TextEditor
+import com.intellij.openapi.options.ShowSettingsUtil
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.ui.EditorNotificationPanel
+import com.intellij.ui.EditorNotifications
+import com.jetbrains.snakecharm.SnakemakeBundle
+
+/**
+ * Shows [EditorNotificationPanel] if 'Enable Snakemake support' setting is disabled
+ */
+class SmkDisabledBanner : EditorNotifications.Provider<EditorNotificationPanel>() {
+
+    private companion object {
+        val KEY = Key.create<EditorNotificationPanel>("smk.content.notification.panel")
+        val DISABLE_NOTIFICATION = Key.create<Boolean>("smk.notification.disabled.hide")
+        var hideNotification = false
+    }
+
+    override fun getKey(): Key<EditorNotificationPanel> = KEY
+
+    override fun createNotificationPanel(
+        file: VirtualFile,
+        fileEditor: FileEditor,
+        project: Project
+    ): EditorNotificationPanel? {
+        val editor = (fileEditor as? TextEditor)?.editor ?: return null
+        if (hideNotification || editor.getUserData(DISABLE_NOTIFICATION) == true || PropertiesComponent.getInstance()
+                .isTrueValue(DISABLE_NOTIFICATION.toString())
+        ) {
+            editor.putUserData(DISABLE_NOTIFICATION, false) // IDK how to update EditorNotifications
+            // after SmkFrameworkConfigurableProvider closure, so we just disable it once, and enable it then
+            return null
+        }
+
+        val settings = SmkSupportProjectSettings.getInstance(project)
+        if (settings.snakemakeSupportEnabled) {
+            return null
+        }
+        val panel = EditorNotificationPanel(fileEditor)
+        panel.text = SnakemakeBundle.message("BANNER.msg")
+        panel.icon(AllIcons.General.Warning)
+        panel.createActionLabel(SnakemakeBundle.message("notifier.msg.framework.by.snakefile.action.configure")) {
+            editor.putUserData(DISABLE_NOTIFICATION, true)
+            ShowSettingsUtil.getInstance().showSettingsDialog(
+                project,
+                SmkFrameworkConfigurableProvider::class.java
+            )
+            EditorNotifications.getInstance(project).updateNotifications(file)
+        }
+        panel.createActionLabel(SnakemakeBundle.message("BANNER.hide")) {
+            // Hides notification for current session
+            hideNotification = true
+            EditorNotifications.getInstance(project).updateNotifications(file)
+        }
+        panel.createActionLabel(SnakemakeBundle.message("BANNER.dont.show.again")) {
+            // Hides notification until cache will be restored
+            PropertiesComponent.getInstance().setValue(DISABLE_NOTIFICATION.toString(), "true")
```

Comment:
Did you investigated where exactly it is stored and how to re-enable it?

----
In you case you save the setting into Application level component (see implementation in AppPropertiesComponentImpl) so check will be disabled not in current project but in all projects which was or will be opened on user machine. Such behaviour is too strict. More over user doesn't have any option how to undo this choice. According to the source code the option is stored in "other.xml" which is located somewhere in PyCharm config system folders, but user will never find it. 

Better solution is to store it per project, i.e. use API call which takes project as argument and implemented in ProjectPropertiesComponentImpl). In such case user most likely will not find where to re-enable it (property is stored in workspace file stored in `[project_dir].idea/workspace.xml` ) but at least it affects only current project and could be reset it user recreate `.idea' folder, e.g. re-create the project 

So I suggest to:
1) store choice per project, and we have own *.xml file with settings, to which SmkSupportProjectSettings are serialized. So use SmkSupportProjectSettings.
2) Not sure that we need place are UI option for renabling it, but at least we need to describe how to re-enable it in github issue, e.g. that user could change an option in ` .idea/snakemake-settings.xml ` 
3) I don't think that we need an option, which disables the check globally for all projects. But for such cases it could be reasonable to use com.intellij.openapi.util.registry.Registry, which is available in `Tools | Internal Actions | Registry...`. Internal actions menu is shown only in dev instance or if to run PyCharm with additional magic options. This case is slightly better than just storing in some unknown file somewhere in system settingsl